File: data/src/main/java/ch/srg/dataProvider/integrationlayer/data/IlHost.java
package ch.srg.dataProvider.integrationlayer.data;

import android.net.Uri;

import androidx.annotation.NonNull;

import java.io.Serializable;

/**
 * Copyright (c) SRG SSR. All rights reserved.
 * <p>
 * License information is available from the LICENSE file.
 */
public final class IlHost implements Serializable {
    public static final IlHost PROD = new IlHost("PROD", "il.srgssr.ch");
    public static final IlHost TEST = new IlHost("TEST", "il-test.srgssr.ch");
    public static final IlHost STAGE = new IlHost("STAGE", "il-stage.srgssr.ch");
    public static final IlHost MMF = new IlHost("MMF", "play-mmf.herokuapp.com/android_26CE9E49-9600");
    public static final IlHost MMF_PUBLIC = new IlHost("MMF", "play-mmf.herokuapp.com");
    public static final IlHost PROD_SAM = new IlHost("PROD_SAM", "il.srgssr.ch/sam");
    public static final IlHost TEST_SAM = new IlHost("TEST_SAM", "il-test.srgssr.ch/sam");
    public static final IlHost STAGE_SAM = new IlHost("STAGE_SAM", "il-stage.srgssr.ch/sam");

    @NonNull
    private final String value;
    @NonNull
    private final String name;

    /**
     * @param name  designed name
     * @param value hostname of the integration layer url
     */
    public IlHost(@NonNull final String name, @NonNull final String value) {
        this.name = name;
        this.value = value;
    }

    @NonNull
    public String getValue() {
        return value;
    }

    @NonNull
    public String getName() {
        return name;
    }

    @NonNull
    public Uri getHostUri() {
        return Uri.parse("https://" + value);
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        IlHost ilHost = (IlHost) o;

        if (!value.equals(ilHost.value)) return false;
        return name.equals(ilHost.name);
    }

    @Override
    public int hashCode() {
        int result = value.hashCode();
        result = 31 * result + name.hashCode();
        return result;
    }

    @NonNull
    @Override
    public String toString() {
        return name;
    }
}

File: dataprovider-retrofit/src/main/java/ch/srg/dataProvider/integrationlayer/request/IlHost.kt
package ch.srg.dataProvider.integrationlayer.request

import android.net.Uri
import java.io.Serializable

/**
 * Copyright (c) SRG SSR. All rights reserved.
 * <p>
 * License information is available from the LICENSE file.
 *
 * @property name  designed name
 * @property value hostname of the integration layer url
 */
data class IlHost(
    val name: String,
    val value: String,
) : Serializable {
    val hostUri: Uri
        get() = Uri.parse("https://$value")

    override fun toString(): String {
        return name
    }

    companion object {
        private const val serialVersionUID = 1L

        @JvmField
        val PROD = IlHost("PROD", "il.srgssr.ch")

        @JvmField
        val TEST = IlHost("TEST", "il-test.srgssr.ch")

        @JvmField
        val STAGE = IlHost("STAGE", "il-stage.srgssr.ch")

        @JvmField
        val MMF = IlHost("MMF", "play-mmf.herokuapp.com/android_26CE9E49-9600")

        @JvmField
        val MMF_PUBLIC = IlHost("MMF", "play-mmf.herokuapp.com")

        @JvmField
        val PROD_SAM = IlHost("PROD_SAM", "il.srgssr.ch/sam")

        @JvmField
        val TEST_SAM = IlHost("TEST_SAM", "il-test.srgssr.ch/sam")

        @JvmField
        val STAGE_SAM = IlHost("STAGE_SAM", "il-stage.srgssr.ch/sam")
    }
}
